src/pages/SignUp/SignUp.js

```diff
@@ -1,7 +1,148 @@
 import React, { Component } from 'react';
+import { chkEmail, chkPwd, chkName, chkAddress } from './Validation';
+import '../SignUp/SignUp.scss';
 
 export default class Login extends Component {
+  emailRef = React.createRef();
+  pwRef = React.createRef();
+
+  state = {
+    name: '',
+    email: '',
+    pw: '',
+    address: '',
+  };
+
+  emailInput = email => {
+    this.setState({
+      email: email.target.value,
+    });
+  };
+
+  pwInput = pw => {
+    this.setState({
+      pw: pw.target.value,
+    });
+  };
+
+  nameInput = name => {
+    this.setState({
+      name: name.target.value,
+    });
+  };
+
+  addressInput = address => {
+    this.setState({
+      address: address.target.value,
+    });
+  };
+
+  checkAll = e => {
+    if (chkName(this.state.name) === false) {
+      alert('이름을 입력해주세요.');
+    }
```

Comment:
- 만약 전부 다 틀렸다면 4번의 alert 창이 차례로 계속 뜨는게 기획 의도일까요? 그렇다고 하더라도 어색해서 하나라도 틀리면 다른 이유들은 굳이 경고창 띄우지 않아도 될 거 같습니다.
- state 구조분해 할당 해주시고, false 체크하고 싶으시다면 반대연산자로 간단히 해줄 수 있을 것 같습니다
```suggestion
    if (!chkName(name)) {
      alert('이름을 입력해주세요.');
    }
```

Reply:
> * 만약 전부 다 틀렸다면 4번의 alert 창이 차례로 계속 뜨는게 기획 의도일까요? 그렇다고 하더라도 어색해서 하나라도 틀리면 다른 이유들은 굳이 경고창 띄우지 않아도 될 거 같습니다.
> * state 구조분해 할당 해주시고, false 체크하고 싶으시다면 반대연산자로 간단히 해줄 수 있을 것 같습니다

4번 모두 띄우는 것은 멘토님 의견처럼 불필요하다는 의견에 동의하여 하나의 alert 으로 통일하였고 placeholder 를 추가하였습니다.